extensions/windows-event-log/wel/JSONUtils.cpp

```diff
@@ -59,29 +60,50 @@ rapidjson::Value xmlDocumentToJSON(const pugi::xml_node& node, rapidjson::Docume
   return children;
 }
 
-void simplifiedGenericXmlToJson(const pugi::xml_node& source_node, rapidjson::Value& output_value, rapidjson::Document& allocator_source_document, bool flatten = false) {
+void simplifiedGenericXmlToJson(const pugi::xml_node& source_node,
+    rapidjson::Value& output_value,
+    rapidjson::Document::AllocatorType& allocator,
+    std::optional<std::string> prefix_for_flat_structure) {
```

Comment:
We should document somewhere, that this has value when the format is flattened, and empty when the format is simple json. A comment or a boolean like `const bool is_flattened = prefix_for_flat_structure.has_value();` could both achieve this.